### PR TITLE
[SMTChecker] Using solUnimplementedAssert instead of solAssert

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes https://github.com/ethereum/solidity/issues/3227

### CODE REVIEW

File: libsolidity/formal/SSAVariable.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ SSAVariable::SSAVariable(
 		m_symbolicVar = make_shared<SymbolicBoolVariable>(_decl, _interface);
 	else
 	{
-		solAssert(false, "");
+		solUnimplementedAssert(false, "Type not supported");
```
Comment (axic) [inline libsolidity/formal/SSAVariable.cpp]: Do we have other possible types here?

**Comment by axic** [comment]
I think the assertions in `SMTChecker.cpp` are good to be changed, since they handle the translation from Solidity and show what feature is missing.

However in the internal parts it must only be using `unimplmented` if it is supported by certain parts of the code, while others don't support it yet and will be implemented. If it is for a future type (sort) which we don't even have yet it should remain a regular assertion.

File: libsolidity/formal/SSAVariable.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SSAVariable.cpp]: Hopefully in the future :p

**Comment by leonardoalt** [comment]
Right, makes sense

**Review by chriseth — APPROVED** [review]

**Pull request merged**